pkg/manager/impl/validation/task_validator.go

```diff
@@ -43,6 +42,26 @@ func validateContainer(task core.TaskTemplate, taskConfig runtime.TaskResourceCo
 	return nil
 }
 
+func validatedK8sPodSpec(task *core.TaskTemplate) error {
+	if task.GetK8SPod().PodSpec == nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument, "K8sPod task type targets must specify a non-empty pod spec")
+	}
+	var podSpec v1.PodSpec
+	jsonObj, err := json.Marshal(task.GetK8SPod().PodSpec)
+	if err != nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument, "Failed to json marshal K8sPod task type target with err: %v", err)
+	}
+	if err = json.Unmarshal(jsonObj, &podSpec); err != nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument, "Failed to json unmarshal K8sPod task type target with err: %v", err)
+	}
+	for _, container := range podSpec.Containers {
+		if errs := validation.IsDNS1123Label(container.Name); len(errs) > 0 {
```

Comment:
Why only validate the name? do we have a dependency on that later? or just trying to prevent issues as early as possible? 
Should that entire function move into the compiler now that Pods are first class citizens?
Also should we use a more elaborate [validation function](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go#L143) if we are trying to prevent runtime errors as much as possible?

Reply:
@EngHabu just trying to prevent issues as soon as possible 
good point re: the compiler, will close out this PR in favor of https://github.com/flyteorg/flytepropeller/pull/286

i looked into k8s pod spec validation functions but there wasn't anything servicable since the spec can be incomplete at the time of registration so I'm not sure if the above linked validation function is necessarily useful at the compilation stage